src/js/services/ws.js

```diff
@@ -102,24 +103,28 @@ export default App.extend({
     if (!_TEST_ && this.resources.length) this._subscribe();
   },
 
+  triggerMessage(data, model) {
+    const channel = this.getChannel();
+    if (model) model.trigger('message', data);
+    if (data.resource) channel.trigger(`message:${ data.resource.type }`, data, model);
+    channel.trigger('message', data, model);
+  },
```

Comment:
These don't need `return`s? Or `triggerMessage()` can do multiple `triggers` in one call?

Reply:
yep triggering multiple things on one call.  one on the model.. two on the service.. one that's generic and one that's resource specific.. this is similar to events on a model.. or model events on a collection.. model has change events but also has like `change:foo` events